pkg/sse/stream.go

```diff
@@ -0,0 +1,52 @@
+package sse
+
+import (
+	"net/http"
+	"reflect"
+)
+
+type Stream struct {
+	w    *Writer
+	done chan struct{}
+}
+
+func OpenStream(w http.ResponseWriter, ch interface{}) *Stream {
+	sw := NewWriter(w)
+	w.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
+	w.WriteHeader(200)
+	sw.Flush()
+
+	s := &Stream{w: sw, done: make(chan struct{})}
+
+	chValue := reflect.ValueOf(ch)
+	go func() {
+		defer s.Close()
+		for {
+			if e, ok := chValue.Recv(); ok {
+				if err := s.Send(e.Interface()); err != nil {
+					return
+				}
+			} else {
+				break
+			}
+		}
+	}()
+
+	return s
+}
+
+func (s *Stream) Send(data interface{}) error {
```

Comment:
This method does not appear to be externally useful without a constructor that doesn't start the send loop?